pkg/operator/controllers/imageconfig/image_controller.go

```diff
@@ -119,18 +119,20 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // Switch case to ensure the correct registries are added depending on the cloud environment (Gov or Public cloud)
 func getCloudAwareRegistries(instance *arov1alpha1.Cluster) ([]string, error) {
-	var requiredRegistries []string
+	acrSubdomain := strings.Split(instance.Spec.ACRDomain, ".")[0]
+	var regionalRegistry string
+
 	switch instance.Spec.AZEnvironment {
 	case azureclient.PublicCloud.Environment.Name:
-		requiredRegistries = []string{instance.Spec.ACRDomain, "arosvc." + instance.Spec.Location + ".data." + azure.PublicCloud.ContainerRegistryDNSSuffix}
+		regionalRegistry = fmt.Sprintf("%s.%s.data.%s", acrSubdomain, instance.Spec.Location, azure.PublicCloud.ContainerRegistryDNSSuffix)
 
 	case azureclient.USGovernmentCloud.Environment.Name:
-		requiredRegistries = []string{instance.Spec.ACRDomain, "arosvc." + instance.Spec.Location + ".data." + azure.USGovernmentCloud.ContainerRegistryDNSSuffix}
+		regionalRegistry = fmt.Sprintf("%s.%s.data.%s", acrSubdomain, instance.Spec.Location, azure.USGovernmentCloud.ContainerRegistryDNSSuffix)
 
 	default:
 		return nil, fmt.Errorf("cloud environment %s is not supported", instance.Spec.AZEnvironment)
 	}
-	return requiredRegistries, nil
+	return []string{instance.Spec.ACRDomain, regionalRegistry}, nil
```

Comment:
Using format strings like this can be fragile, but perhaps we can DRY it out a bit to make it a bit more robust.

e.g.
```golang
var regionalRegistrySuffix string
switch instance.Spec.AZEnvironment {
case azureclient.PublicCloud.Environment.Name:
	regionalRegistrySuffix = azure.PublicCloud.ContainerRegistryDNSSuffix
 [snip -- repeat for USGovernmentCloud and maintain default case)

// only have this format string once, because the format string is the fragile piece
regionalRegistry := fmt.Sprintf("%s.%s.data.%s", acrSubdomain, instance.Spec.Location, regionalRegistrySuffix)

return []string{instance.Spec.ACRDomain, regionalRegistry}, nil
```

Reply:
I couldn't find any information about fragility of format strings in Go (if you have any good resources I wouldn't mind reading about it), but I did change this to follow the [Uber style guide](https://github.com/uber-go/guide/blob/master/style.md#format-strings-outside-printf) recommendation to use a const as the string template.